sqlserver/datadog_checks/sqlserver/sqlserver.py

```diff
@@ -134,15 +135,19 @@ def initialize_connection(self):
         except Exception as e:
             self.log.exception("Initialization exception %s", e)
 
-    def handle_service_check(self, status, host, database, message=None):
+    def handle_service_check(self, status, host, database, message=None, is_default=True):
         custom_tags = self.instance.get("tags", [])
         if custom_tags is None:
             custom_tags = []
+
         service_check_tags = ['host:{}'.format(host), 'db:{}'.format(database)]
         service_check_tags.extend(custom_tags)
         service_check_tags = list(set(service_check_tags))
 
-        self.service_check(SERVICE_CHECK_NAME, status, tags=service_check_tags, message=message, raw=True)
+        if self.autodiscovery:
+            self.service_check(DATABASE_SERVICE_CHECK_NAME, status, tags=service_check_tags, message=message, raw=True)
```

Comment:
Can you make sure if the service check is `OK`, there won't be a message submitted? We are no longer adding any messages to `OK` service checks. See more examples of that here: https://github.com/DataDog/integrations-core/pull/9888/files#diff-1f16a4e0adeed42c3800bf657595d9fba1b4fba04393efdf395f415c1bfb4568R144-R146

Reply:
Right now if the status is `OK` I don't send one: https://github.com/DataDog/integrations-core/pull/9900/files#diff-a38a2a09e86a943d0704a9f2922133905324820865ff052eff46209ee989d6d9R160

Do you think we need an extra check in the `handle_service_check` function or is it good enough that it's never called with the message parameter when the status is `OK`?

Reply:
I think it's good enough without an extra check. I added an exception that would be thrown during tests if the condition is broken: https://github.com/DataDog/integrations-core/pull/9898.